app/javascript/react/screens/App/Overview/screens/MappingWizard/components/MappingWizardDatastoresStep/components/helpers.js

```diff
@@ -1,22 +1,65 @@
 import numeral from 'numeral';
 
-const calculateTotalUsedSpace = sourceDatastores =>
-  sourceDatastores.reduce(
-    (totalSpace, sourceDatastore) =>
-      (totalSpace += sourceDatastore.total_space - sourceDatastore.free_space),
+export const datastoreUsedSpace = (datastore = {}) =>
+  datastore.total_space - datastore.free_space;
+
+export const totalUsedSpace = (datastores = []) =>
+  datastores.reduce(
+    (totalSpace, datastore) => (totalSpace += datastoreUsedSpace(datastore)),
     0
   );
 
-const diskSpaceInfo = (targetDatastore, sourceDatastores) => {
+export const targetDatastoreAvailableSpace = (
+  targetDatastore,
+  datastoresStepMappings
+) => {
+  const datastoresMappings = datastoresStepMappings.reduce(
+    (mappings, targetClusterWithDatastoresMappings) =>
+      mappings.concat(targetClusterWithDatastoresMappings.nodes),
```

Comment:
Again you can use `[...a, ...b]` instead of `a.concat(b)` if you want, but not a PR blocker.